src/arch/riscv/RiscvMMU.py

```diff
@@ -49,7 +49,7 @@ class RiscvMMU(BaseMMU):
 
     itb = RiscvTLB(entry_type="instruction")
     dtb = RiscvTLB(entry_type="data")
-    pma_checker = Param.BasePMAChecker(PMAChecker(), "PMA Checker")
+    pma_checker = Param.PMAChecker("PMA Checker")
```

Comment:
I strongly agree with this change to remove an instance of a SimObject as the default value for a parameter.

However, I worry that there are configuration scripts that depend on this. We need to test the stdlib boards and the config scripts in `configs/examples` before we can merge this change.

In fact, it looks like this is likely why the CI tests are failing. See https://github.com/gem5/gem5/actions/runs/10286529082/job/28476297182?pr=1401